_posts/2021-03-25-sparklyr-1.6.0-released/sparklyr-1.6.0.Rmd

```diff
@@ -0,0 +1,450 @@
+author:
+  - name: Yitao Li
+    url: https://github.com/yitao-li
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com
+slug: sparklyr-1.6
+date: 03-25-2021
+bibliography: bibliography.bib
+categories:
+  - R
+  - Packages/Releases
+  - Distributed Computing
+output:
+  distill::distill_article:
+    self_contained: false
+preview: images/sparklyr-1.6.jpg
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = FALSE)
+```
+
+[`Sparklyr`](https://sparklyr.ai) 1.6 is now available on [CRAN](https://cran.r-project.org/web/packages/sparklyr/index.html)!
+
+To install `sparklyr` 1.6 from CRAN, run
+
+```{r echo=TRUE, eval=FALSE}
+install.packages("sparklyr")
+```
+
+In this blog post, we shall highlight the following features and enhancements
+from `sparklyr` 1.6:
+
+ * [Weighted quantile summaries](#weighted-quantile-summaries)
+ * [Power iteration clustering](#power-iteration-clustering)
+ * [`spark_write_rds()` + `collect_from_rds()`](#spark_write_rds-collect_from_rds)
+ * [Dplyr-related improvements](#dplyr-related-improvements)
+
+## Weighted quantile summaries
+
+[Apache Spark](https://spark.apache.org) is well-known for supporting
+approximate algorithms that trade off marginal amount of accuracy for greater
+speed and parallelism.
+Such algorithms are particularly beneficial for performing preliminary data
+explorations at scale, as they enable users to quickly query certain estimated
+statistics within a predefined error margin while avoiding the high cost of
+exact computations.
+One such algorithm is the Greenwald-Khanna online computation of quantile
+summaries, as described in @10.1145/376284.375670.
+While this algorithm was originally designed for efficient $\epsilon$-
+approximation of quantiles within a large dataset without the notion of data
+points carrying different weights, and the unweighted version of it has been
+implemented as
+[`approxQuantile()`](https://spark.apache.org/docs/2.0.0/api/java/org/apache/spark/sql/DataFrameStatFunctions.html#approxQuantile(java.lang.String,%20double[],%20double))
+ since Spark 2.0, the same algorithm can be generalized to handle weighted
+inputs, and as `sparklyr` user [\@Zhuk66](https://github.com/Zhuk66) mentioned
+in [this issue](https://github.com/sparklyr/sparklyr/issues/2915), a
+[weighted version](https://github.com/sparklyr/sparklyr/blob/4b6bc6677ecf92787ab3521f364a8d80b973d92f/java/spark-1.5.2/weightedquantilesummaries.scala#L13-L332)
+of this algorithm can be a useful `sparklyr` feature.
+
+To properly explain what weighted-quantile means, we must clarify what the
+weight of each data point signifies. For example, if we have a sequence of
+observations $(1, 1, 1, 1, 0, 2, -1, -1)$ and would like to find the median
+of all data points, then we can either run the unweighted version of
```

Comment:
tried to simplify this sentence but don't have a good idea ;-) how about creating two bullet points instead, corresponding to the either-or?